### PR TITLE
fix(gqa): size present KV init to past + current seq length

### PR DESCRIPTION
**Pull request by BoarQing**
## Summary

`GroupQueryAttentionToHip` sized the `present_key` / `present_value` DPS init tensors from the **past** operand alone. Present KV is `concat(past, current)` along the seq axis, so its seq extent is `past_seq + current_seq`. On a fresh prefill `past_seq == 0`, so the init collapsed to a **zero-length buffer**; the output allocator then hands the runtime a null `present_key`/`present_value`, `wrap_group_query_attention` rejects the call, and the attention output is left zero-filled.

On gpt-oss-20b this skips *every* GQA call in the graph (24 layers), and the model emits gibberish.

This ports the `%tot = %past + %cur` rule that `OnnxAttentionConversion.cpp` already applies to `onnx.Attention`.

## Root cause / when it was introduced

- Introduced in db7c51fc, "Port LLaMA operations from PR-17: rope, matmul, gqa, gather" (#34, 2026-03-15) — the very first GQA conversion, as `createEmptyTensor(rewriter, loc, presentKeyType, pastKey)`.
- #62 ("GQA: Complete MS Specification Alignment") later changed it to `pastKey ? pastKey : (key ? key : query)`, which only added a null fallback and kept the wrong dimension source.
- #98 moved the code into `GqaConversion.cpp` during the monolith split.

So this is a day-one bug in the GQA path, not a recent regression.

## Why lit never caught it

Every existing GQA lit test (`test_gqa.mlir`, `test_gqa_prefill_sentinel.mlir`) uses **fully static** present shapes. With a static result type the init is a plain `tensor.empty()` with no dynamic operands, so the sizing source is unobservable and the buggy and correct code emit identical IR. The bug only shows up when the present seq dim is dynamic, which is what the real gpt-oss export produces.

This PR adds `test_gqa_dynamic_present.mlir` with dynamic batch/seq. Verified it **fails** against a pre-fix `hip-mlir-opt` and **passes** after the fix.

Pre-fix IR (present sized from `%arg4` = `past_key` only, no `addi`):

```mlir
%dim_1 = tensor.dim %arg4, %c0 : tensor<?x8x?x128xf16>
%dim_2 = tensor.dim %arg4, %c2 : tensor<?x8x?x128xf16>
%1 = tensor.empty(%dim_1, %dim_2) : tensor<?x8x?x128xf16>   // past_seq == 0 -> empty
```

Post-fix IR:

```mlir
%dim_2 = tensor.dim %arg1, %c1 : tensor<?x?x4096xf16>       // current
%dim_3 = tensor.dim %arg4, %c2 : tensor<?x8x?x128xf16>      // past
%1 = arith.addi %dim_3, %dim_2 : index                      // total
%2 = tensor.empty(%dim_1, %1) : tensor<?x8x?x128xf16>
```

## Scope guard

Only rank-3 BSH query (seq at dim 1) with rank-4 BNSH present (seq at dim 2) is re-sized. Any other layout falls back to the previous positional behaviour rather than guessing which axis carries the sequence.

## Validation

Local gpt-oss-20b greedy decode, 125-token prompt, 128 new tokens, HipEP on gfx1151:

| | before | after |
|---|---|---|
| `GQA requires present_key/present_value` errors | 3072 (24 layers x 128 steps) | 0 |
| first token | `11` (`,`) | `20837` (`' AI'`) — matches CPU EP exactly |
| output | `, and la) (H] (H] (H] (H]...` | ` AI, fully or partially or fully?\n\nAMD has fully embraced generative AI...` |
| prefill logits cosine vs CPU EP | — | 0.99998 |

Lit: all `onnx-to-hip` conversion tests pass (107 pass; 4 negative tests need `not`, 2 need `%t`, both limitations of my ad-hoc local runner, not the tests).

## Test plan

- [ ] CI lit: `test/lit/Conversion/onnx-to-hip/test_gqa_dynamic_present.mlir`
- [ ] CI accuracy/E2E on gpt-oss-20b (this is the run to watch — please confirm the CI gibberish clears)
- [ ] Existing GQA/Attention conversion tests still green

### CODE REVIEW

**Comment by BoarQing** [comment]
Closing: this change is wrong for the contract hip-ep actually supports.

`test/lit/e2e/test_gqa_output_allocator_present_dim.mlir` (added in #306) deliberately pins present.* to be sized from the matching past input's extent. Under OGA `past_present_share_buffer` that extent is the max_length capacity buffer, and `output_allocate_cb` passes the in-graph shape to `GetOutput` verbatim specifically so ORT returns the pre-bound OrtValue and the past==present pointer identity is preserved. Sizing present to `past + current` makes the requested shape `max_length + current`, which no longer matches the pre-bound value, so the KV-cache aliasing would be lost.

The empty-past case that motivated this only occurs with `past_present_share_buffer: false`, which OGA never uses, so this was never going to affect the gpt-oss-20b CI accuracy run either. Supporting the non-shared-buffer contract would be a deliberate change to the allocator invariant, not this patch.
